Warn instead of error on missing entities.png

alexander-b opened the pull request:
Signed-off-by: Alexander Berntsen alexander@plaimi.net

alexander-b commented:
Fixes #884.

christopho commented:
If we decide to make it optional, there should be no error/warning at all.

alexander-b commented:
Here is a commit that only complains when it's being used.

christopho commented on changelog.txt:
```diff
@@ -103,6 +103,7 @@ Changes that do not introduce incompatibilities:
 * Add methods sprite:get_size() and sprite:get_origin() (#823).
 * sprite:set_animation() now takes an optional callback parameter (#861).
 * Add a method surface:get_opacity() (#722).
+* Make "tileset.entities.png" files optional (#884).
```
This should be in the "Data file format changes" section just below.

christopho commented on src/entities/Tileset.cpp:
```diff
@@ -194,6 +193,9 @@ const SurfacePtr& Tileset::get_tiles_image() {
  */
 const SurfacePtr& Tileset::get_entities_image() {
+  if (entities_image == nullptr) {
+    Debug::error(std::string("Missing entities image for tileset '") + id + "': " + std::string("tilesets/") + id + ".dat" );
```
This will complain too many times I think. (Every time the image is requested)

alexander-b commented:
I thought it might be useful to whine every time it would be accessed. I modified it now to only whine when set_tileset() is called. I also fixed moved changelog entry.

christopho commented:
Closing this pull request. The entities.png file is now optional and there is no error or warning (see 6a02a21).
